packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/listFeatures.ts

```diff
@@ -456,6 +459,83 @@ function shouldTriggerList(
     );
 }
 
+/**
+ * MergeListOnBackspaceAfterList edit feature, provides the ability to merge list on backspace on block after a list.
+ */
+const MergeListOnBackspaceAfterList: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.BACKSPACE],
+    shouldHandleEvent: (event, editor) => {
+        const target = editor.getElementAtCursor();
+        if (target) {
+            const cursorBlock = editor.getBlockElementAtNode(target)?.getStartNode() as HTMLElement;
+            const previousBlock = cursorBlock?.previousElementSibling ?? null;
+
+            if (isList(previousBlock)) {
+                const range = editor.getSelectionRange();
+                const searcher = editor.getContentSearcherOfCursor(event);
+                const textBeforeCursor = searcher?.getSubStringBefore(4);
+                const nearestInline = searcher?.getNearestNonTextInlineElement();
+
+                if (range && range.collapsed && textBeforeCursor === '' && !nearestInline) {
+                    const tempBlock = cursorBlock?.nextElementSibling;
+                    const nextBlock = isList(tempBlock) ? tempBlock : tempBlock?.firstChild;
+
+                    if (isList(nextBlock)) {
+                        const element = cacheGetEventData<HTMLOListElement | HTMLUListElement>(
+                            event,
+                            'previousBlock',
+                            () => previousBlock
+                        );
+                        const nextElement = cacheGetEventData<HTMLOListElement | HTMLUListElement>(
+                            event,
+                            'nextBlock',
+                            () => nextBlock
+                        );
+
+                        return !!element && !!nextElement;
+                    }
+                }
+            }
+        }
+
+        return false;
+    },
+    handleEvent: (event, editor) => {
+        editor.runAsync(editor => {
+            const previousList = cacheGetEventData<HTMLOListElement | HTMLUListElement | null>(
+                event,
+                'previousBlock',
+                () => null
+            );
+            const targetBlock = cacheGetEventData<HTMLOListElement | HTMLUListElement | null>(
+                event,
+                'nextBlock',
+                () => null
+            );
+
+            const rangeBeforeWriteBack = editor.getSelectionRange();
+
+            if (previousList && targetBlock && rangeBeforeWriteBack) {
+                const fvList = new VList(previousList);
+                fvList.mergeVList(new VList(targetBlock));
```

Comment:
What happens if try to merge a two different types of list?

Reply:
Testing this scenario right now in Rooster it is not merging the List with different Tag Names, but I just added another check in the ShouldHandleEvent Callback to avoid that happening.